pkg/jmxfetch/jmxfetch.go

```diff
@@ -496,3 +511,24 @@ func (j *JMXFetch) ConfigureFromInstance(instance integration.Data) error {
 
 	return nil
 }
+
+func (j *JMXFetch) getDSDStatus() DSDStatus {
+	// Three possible states: DSD is running in the Core Agent, DSD is running via ADP, or the DSD status is unknown.
+	//
+	// We detect these through the `use_dogstatsd` configuration and the `DD_ADP_ENABLED` environment variable, and we
+	// detect whether or not we're listening on UDS or UDP via the configuration settings that define their listening
+	// address.
+	dsdEnabledInternally := pkgconfigsetup.Datadog().GetBool("use_dogstatsd")
+	adpEnabled := os.Getenv("DD_ADP_ENABLED") == "true"
+	dsdEnabled := dsdEnabledInternally || adpEnabled
+	udsEnabled := pkgconfigsetup.Datadog().GetString("dogstatsd_socket") != ""
+	udpEnabled := pkgconfigsetup.Datadog().GetInt("dogstatsd_port") != 0
+
+	if dsdEnabled && udsEnabled {
+		return DSDStatusRunningUDSDatagram
+	} else if dsdEnabled && udpEnabled {
+		return DSDStatusRunningUDP
+	} else {
```

Comment:
What happens when you have both `dogstatsd_socket` and `dogstatsd_port` ? Does UDS have priority ?

Reply:
Yep, which maps to the original behavior where we always preferred UDS over UDP.